docs/design/changelog-management-system.md

```diff
@@ -0,0 +1,480 @@
+        "Features Added",
+        "Breaking Changes",
+        "Bugs Fixed",
+        "Other Changes"
+      ]
+    },
+    "subsection": {
+      "type": ["string", "null"]
+    },
+    "description": {
+      "type": "string",
+      "minLength": 10
+    },
+    "pr": {
+      "type": "integer",
+      "minimum": 1
+    }
+  },
+  "additionalProperties": false
+}
+```
+
+### 2. Generator Script
+
+**File:** `eng/scripts/New-ChangelogEntry.ps1`
+
+Helper script to create properly formatted changelog entries.
+
+**Usage:**
+
+```powershell
+# Interactive mode (prompts for all fields)
+./eng/scripts/New-ChangelogEntry.ps1
+
+# With parameters
+./eng/scripts/New-ChangelogEntry.ps1 `
+  -Description "Added new feature" `
+  -Section "Features Added" `
+  -PR 1234
+
+# With subsection
+./eng/scripts/New-ChangelogEntry.ps1 `
+  -Description "Updated Azure.Core to 1.2.3" `
+  -Section "Other Changes" `
+  -Subsection "Dependency Updates" `
+  -PR 1234
+```
+
+**Features:**
+- Interactive prompts for section, description, PR number
+- Auto-generates filename with timestamp
+- Validates YAML against schema
+- Places file in correct directory
+- Creates `changelog-entries/` directory if it doesn't exist
+
+### 3. Compiler Script
+
+**File:** `eng/scripts/Compile-Changelog.ps1`
+
+Compiles all YAML entries into CHANGELOG.md.
+
+**Usage:**
+
+```powershell
+# Preview what will be compiled (dry run)
+./eng/scripts/Compile-Changelog.ps1 -DryRun
+
+# Compile entries into CHANGELOG.md
+./eng/scripts/Compile-Changelog.ps1
+
+# Compile and remove YAML files after successful compilation
+./eng/scripts/Compile-Changelog.ps1 -DeleteFiles
+
+# Custom changelog path
+./eng/scripts/Compile-Changelog.ps1 -ChangelogPath "path/to/CHANGELOG.md"
+```
+
+**Features:**
+- Reads all YAML files from `changelog-entries/`
+- Validates each file against schema
+- Groups entries by section, then by subsection
+- Formats entries as markdown with PR links
+- Inserts compiled entries into CHANGELOG.md under "Unreleased"
+- Optional deletion of YAML files after compilation
+- Error handling for missing/invalid files
+- Summary output
+
+**Compilation Logic:**
+
+1. Read all `.yml` files from `changelog-entries/` (excluding README.md)
+2. Validate each file against schema
+3. Group by section (preserving order: Features, Breaking, Bugs, Other)
+4. Within each section, group by subsection (if present)
+5. Sort entries within groups
+6. Generate markdown format with PR links
+7. Find the "Unreleased" section in CHANGELOG.md
+8. Insert compiled entries under appropriate section headers
+9. Optionally delete YAML files if `-DeleteFiles` flag is set
+
+---
+
+## Example Workflow
+
+### For Contributors
+
+When making a change that needs a changelog entry:
+
+1. **Make your code changes**
+
+2. **Create a changelog entry:**
+   ```powershell
+   ./eng/scripts/New-ChangelogEntry.ps1 `
+     -Description "Your change description" `
+     -Section "Features Added" `
+     -PR <pr-number>
```

Comment:
Should we also include the contributor in some way? https://github.com/microsoft/mcp/issues/156

Reply:
The contributor can be found in the linked PR and doesn't have to be explicitly written in the change log, isn't it?